# Profitable App Profiles for the App Store and Google Play Markets

- We only build free apps and our revenue consists of in-app ads. Therefore, our revenue for any given app is mostly influenced by the number of users.
- Goal for this project is to analyze data to help our developers understand what type of apps are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

### Dataset
- A data set containing data about approximately 10,000 Android apps from Google Play; the data was collected in August 2018. 
- A data set containing data about approximately 7,000 iOS apps from the App Store; the data was collected in July 2017. 

### Opening and Exploring Datasets



In [1]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

        
# dataset: list of lists
# start, end: integers, staring and the ending indices of a slice from dataset


In [2]:
file = open('/Users/kitaeklee/Desktop/Data/AppStore/AppleStore.csv')

from csv import reader
read_file = reader(file)
data_list = list(read_file)


In [3]:
file2 = open('/Users/kitaeklee/Desktop/Data/AppStore/googleplaystore.csv')

from csv import reader
read_file2 = reader(file2)
data_list2 = list(read_file2)

In [4]:
explore_data(data_list, 1, 3)

['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']




In [5]:
explore_data(data_list2, 1, 3)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']




In [6]:
# column names of first dataset
explore_data(data_list, 0, 1)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']




In [7]:
# column names of second dataset
explore_data(data_list2, 0, 1)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']




# Data Cleaning

##### Apple Store Dataset

In [8]:
import pandas as pd

# Create dataframe using list of lists of the first dataset
df = pd.DataFrame(data_list)
header = ['','id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
df = df[1:]
df.columns = header
df = df.drop(df.columns[0], axis=1)

df

id                                         track_name  \
1      281656475                                    PAC-MAN Premium   
2      281796108                          Evernote - stay organized   
3      281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   
4      282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   
5      282935706                                              Bible   
...          ...                                                ...   
7193  1187617475                                              Kubik   
7194  1187682390                                  VR Roller-Coaster   
7195  1187779532              Bret Michaels Emojis + Lyric Keyboard   
7196  1187838770          VR Roller Coaster World - Virtual Reality   
7197  1188375727                       Escape the Sweet Shop Series   

     size_bytes currency price rating_count_tot rating_count_ver user_rating  \
1     100788224      USD  3.99            21292               26           4   
2     158578688      USD     0           161065               26           4   
3     100524032      USD     0           188583             2822         3.5   
4     128512000      USD     0           262241              649           4   
5      92774400      USD     0           985920             5320         4.5   
...         ...      ...   ...              ...              ...         ...   
7193  126644224      USD     0              142               75         4.5   
7194  120760320      USD     0               30               30         4.5   
7195  111322112      USD  1.99               15                0         4.5   
7196   97235968      USD     0               85               32         4.5   
7197   90898432      USD     0                3                3           5   

     user_rating_ver     ver cont_rating   prime_genre sup_devices.num  \
1                4.5   6.3.5          4+         Games              38   
2                3.5   8.2.2          4+  Productivity              37   
3                4.5   5.0.0          4+       Weather              37   
4                4.5  5.10.0         12+      Shopping              37   
5                  5   7.5.1          4+     Reference              37   
...              ...     ...         ...           ...             ...   
7193             4.5     1.3          4+         Games              38   
7194             4.5     0.9          4+         Games              38   
7195               0   1.0.2          9+     Utilities              37   
7196             4.5  1.0.15         12+         Games              38   
7197               5     1.0          4+         Games              40   

     ipadSc_urls.num lang.num vpp_lic  
1                  5       10       1  
2                  5       23       1  
3                  5        3       1  
4                  5        9       1  
5                  5       45       1  
...              ...      ...     ...  
7193               5        1       1  
7194               0        1       1  
7195               1        1       1  
7196               0        2       1  
7197               0        2       1  

[7197 rows x 16 columns]

Check the number of duplicates that have the same name of applications.

In [9]:
# Number of duplicates in the AppStore dataset
duplicate_list=[]
for row in df.duplicated(subset='track_name'):
    if row == True:
        duplicate_list.append(row)
num = len(duplicate_list)
print("Number of duplicates:", num)


Number of duplicates: 2


There are 2 duplicates therefore, remove the duplicates based on the name of the application but keep the first one.

In [10]:
# Drop duplicates
df.drop_duplicates(subset='track_name', keep='first', inplace=True)

df

id                                         track_name  \
1      281656475                                    PAC-MAN Premium   
2      281796108                          Evernote - stay organized   
3      281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   
4      282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   
5      282935706                                              Bible   
...          ...                                                ...   
7193  1187617475                                              Kubik   
7194  1187682390                                  VR Roller-Coaster   
7195  1187779532              Bret Michaels Emojis + Lyric Keyboard   
7196  1187838770          VR Roller Coaster World - Virtual Reality   
7197  1188375727                       Escape the Sweet Shop Series   

     size_bytes currency price rating_count_tot rating_count_ver user_rating  \
1     100788224      USD  3.99            21292               26           4   
2     158578688      USD     0           161065               26           4   
3     100524032      USD     0           188583             2822         3.5   
4     128512000      USD     0           262241              649           4   
5      92774400      USD     0           985920             5320         4.5   
...         ...      ...   ...              ...              ...         ...   
7193  126644224      USD     0              142               75         4.5   
7194  120760320      USD     0               30               30         4.5   
7195  111322112      USD  1.99               15                0         4.5   
7196   97235968      USD     0               85               32         4.5   
7197   90898432      USD     0                3                3           5   

     user_rating_ver     ver cont_rating   prime_genre sup_devices.num  \
1                4.5   6.3.5          4+         Games              38   
2                3.5   8.2.2          4+  Productivity              37   
3                4.5   5.0.0          4+       Weather              37   
4                4.5  5.10.0         12+      Shopping              37   
5                  5   7.5.1          4+     Reference              37   
...              ...     ...         ...           ...             ...   
7193             4.5     1.3          4+         Games              38   
7194             4.5     0.9          4+         Games              38   
7195               0   1.0.2          9+     Utilities              37   
7196             4.5  1.0.15         12+         Games              38   
7197               5     1.0          4+         Games              40   

     ipadSc_urls.num lang.num vpp_lic  
1                  5       10       1  
2                  5       23       1  
3                  5        3       1  
4                  5        9       1  
5                  5       45       1  
...              ...      ...     ...  
7193               5        1       1  
7194               0        1       1  
7195               1        1       1  
7196               0        2       1  
7197               0        2       1  

[7195 rows x 16 columns]

Our firm only build apps that are free to download and install. Therefore, recreate the dataframe only with the apps that are free.

In [11]:
# Drop the rows with non-free apps since we only deal with free apps.
df = df[df['price'] == '0']

In [12]:
# Drop rows with missing values just in case
df.dropna(axis=0, inplace=True)

/Users/kitaeklee/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [13]:
# Reset the index
df.reset_index(drop=True, inplace=True)
df

id                                         track_name  \
0      281796108                          Evernote - stay organized   
1      281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   
2      282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   
3      282935706                                              Bible   
4      283646709             PayPal - Send and request money safely   
...          ...                                                ...   
4049  1186384912       Demolition Derby Virtual Reality (VR) Racing   
4050  1187617475                                              Kubik   
4051  1187682390                                  VR Roller-Coaster   
4052  1187838770          VR Roller Coaster World - Virtual Reality   
4053  1188375727                       Escape the Sweet Shop Series   

     size_bytes currency price rating_count_tot rating_count_ver user_rating  \
0     158578688      USD     0           161065               26           4   
1     100524032      USD     0           188583             2822         3.5   
2     128512000      USD     0           262241              649           4   
3      92774400      USD     0           985920             5320         4.5   
4     227795968      USD     0           119487              879           4   
...         ...      ...   ...              ...              ...         ...   
4049  168774656      USD     0               18               18           4   
4050  126644224      USD     0              142               75         4.5   
4051  120760320      USD     0               30               30         4.5   
4052   97235968      USD     0               85               32         4.5   
4053   90898432      USD     0                3                3           5   

     user_rating_ver     ver cont_rating   prime_genre sup_devices.num  \
0                3.5   8.2.2          4+  Productivity              37   
1                4.5   5.0.0          4+       Weather              37   
2                4.5  5.10.0         12+      Shopping              37   
3                  5   7.5.1          4+     Reference              37   
4                4.5  6.12.0          4+       Finance              37   
...              ...     ...         ...           ...             ...   
4049               4   1.0.0         12+         Games              38   
4050             4.5     1.3          4+         Games              38   
4051             4.5     0.9          4+         Games              38   
4052             4.5  1.0.15         12+         Games              38   
4053               5     1.0          4+         Games              40   

     ipadSc_urls.num lang.num vpp_lic  
0                  5       23       1  
1                  5        3       1  
2                  5        9       1  
3                  5       45       1  
4                  0       19       1  
...              ...      ...     ...  
4049               4        1       1  
4050               5        1       1  
4051               0        1       1  
4052               0        2       1  
4053               0        2       1  

[4054 rows x 16 columns]

##### Google Playstore Dataset

In [14]:
# Create a dataframe using the list of lists of the second dataset
df2 = pd.DataFrame(data_list2)
df2 = df2[1:]
header2 = ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
df2.columns = header2


df2

App             Category  \
1         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
2                                    Coloring book moana       ART_AND_DESIGN   
3      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
4                                  Sketch - Draw & Paint       ART_AND_DESIGN   
5                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10837                                   Sya9a Maroc - FR               FAMILY   
10838                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10839                             Parkinson Exercices FR              MEDICAL   
10840                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10841      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

      Rating Reviews                Size     Installs  Type Price  \
1        4.1     159                 19M      10,000+  Free     0   
2        3.9     967                 14M     500,000+  Free     0   
3        4.7   87510                8.7M   5,000,000+  Free     0   
4        4.5  215644                 25M  50,000,000+  Free     0   
5        4.3     967                2.8M     100,000+  Free     0   
...      ...     ...                 ...          ...   ...   ...   
10837    4.5      38                 53M       5,000+  Free     0   
10838    5.0       4                3.6M         100+  Free     0   
10839    NaN       3                9.5M       1,000+  Free     0   
10840    4.5     114  Varies with device       1,000+  Free     0   
10841    4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
1           Everyone               Art & Design   January 7, 2018   
2           Everyone  Art & Design;Pretend Play  January 15, 2018   
3           Everyone               Art & Design    August 1, 2018   
4               Teen               Art & Design      June 8, 2018   
5           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10837       Everyone                  Education     July 25, 2017   
10838       Everyone                  Education      July 6, 2018   
10839       Everyone                    Medical  January 20, 2017   
10840     Mature 17+          Books & Reference  January 19, 2015   
10841       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
1                   1.0.0        4.0.3 and up  
2                   2.0.0        4.0.3 and up  
3                   1.2.4        4.0.3 and up  
4      Varies with device          4.2 and up  
5                     1.1          4.4 and up  
...                   ...                 ...  
10837                1.48          4.1 and up  
10838                 1.0          4.1 and up  
10839                 1.0          2.2 and up  
10840  Varies with device  Varies with device  
10841  Varies with device  Varies with device  

[10841 rows x 13 columns]

Based on discussion on Kaggle, there is missing value in one row.

In [15]:
# Based on discussion on Kaggle, there is missing value.
df2.loc[10473, :]

App               Life Made WI-Fi Touchscreen Photo Frame
Category                                              1.9
Rating                                                 19
Reviews                                              3.0M
Size                                               1,000+
Installs                                             Free
Type                                                    0
Price                                            Everyone
Content Rating                                           
Genres                                  February 11, 2018
Last Updated                                       1.0.19
Current Ver                                    4.0 and up
Android Ver                                          None
Name: 10473, dtype: object

There is missing value in Content Rating. Therefore, drop that row.

In [16]:
# Drop rows with missing values
df2.dropna(axis=0, inplace=True)
df2

App             Category  \
1         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
2                                    Coloring book moana       ART_AND_DESIGN   
3      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
4                                  Sketch - Draw & Paint       ART_AND_DESIGN   
5                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10837                                   Sya9a Maroc - FR               FAMILY   
10838                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10839                             Parkinson Exercices FR              MEDICAL   
10840                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10841      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

      Rating Reviews                Size     Installs  Type Price  \
1        4.1     159                 19M      10,000+  Free     0   
2        3.9     967                 14M     500,000+  Free     0   
3        4.7   87510                8.7M   5,000,000+  Free     0   
4        4.5  215644                 25M  50,000,000+  Free     0   
5        4.3     967                2.8M     100,000+  Free     0   
...      ...     ...                 ...          ...   ...   ...   
10837    4.5      38                 53M       5,000+  Free     0   
10838    5.0       4                3.6M         100+  Free     0   
10839    NaN       3                9.5M       1,000+  Free     0   
10840    4.5     114  Varies with device       1,000+  Free     0   
10841    4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
1           Everyone               Art & Design   January 7, 2018   
2           Everyone  Art & Design;Pretend Play  January 15, 2018   
3           Everyone               Art & Design    August 1, 2018   
4               Teen               Art & Design      June 8, 2018   
5           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10837       Everyone                  Education     July 25, 2017   
10838       Everyone                  Education      July 6, 2018   
10839       Everyone                    Medical  January 20, 2017   
10840     Mature 17+          Books & Reference  January 19, 2015   
10841       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
1                   1.0.0        4.0.3 and up  
2                   2.0.0        4.0.3 and up  
3                   1.2.4        4.0.3 and up  
4      Varies with device          4.2 and up  
5                     1.1          4.4 and up  
...                   ...                 ...  
10837                1.48          4.1 and up  
10838                 1.0          4.1 and up  
10839                 1.0          2.2 and up  
10840  Varies with device  Varies with device  
10841  Varies with device  Varies with device  

[10840 rows x 13 columns]

Check the number of duplicates that have the same name of the application.

In [17]:
# Number of duplicates in the Googlestore dataset
duplicate_list2=[]
for row in df2.duplicated(subset='App'):
    if row == True:
        duplicate_list2.append(row)
num2 = len(duplicate_list2)
print("Number of duplicates:", num2)

Number of duplicates: 1181


There are 1181 duplicates. Drop the duplicates based on the name of the applications while keeping the first duplicate.

In [18]:
# Drop duplicates
df2.drop_duplicates(subset='App', keep='first', inplace=True)

df2

App             Category  \
1         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
2                                    Coloring book moana       ART_AND_DESIGN   
3      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
4                                  Sketch - Draw & Paint       ART_AND_DESIGN   
5                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10837                                   Sya9a Maroc - FR               FAMILY   
10838                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10839                             Parkinson Exercices FR              MEDICAL   
10840                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10841      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

      Rating Reviews                Size     Installs  Type Price  \
1        4.1     159                 19M      10,000+  Free     0   
2        3.9     967                 14M     500,000+  Free     0   
3        4.7   87510                8.7M   5,000,000+  Free     0   
4        4.5  215644                 25M  50,000,000+  Free     0   
5        4.3     967                2.8M     100,000+  Free     0   
...      ...     ...                 ...          ...   ...   ...   
10837    4.5      38                 53M       5,000+  Free     0   
10838    5.0       4                3.6M         100+  Free     0   
10839    NaN       3                9.5M       1,000+  Free     0   
10840    4.5     114  Varies with device       1,000+  Free     0   
10841    4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
1           Everyone               Art & Design   January 7, 2018   
2           Everyone  Art & Design;Pretend Play  January 15, 2018   
3           Everyone               Art & Design    August 1, 2018   
4               Teen               Art & Design      June 8, 2018   
5           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10837       Everyone                  Education     July 25, 2017   
10838       Everyone                  Education      July 6, 2018   
10839       Everyone                    Medical  January 20, 2017   
10840     Mature 17+          Books & Reference  January 19, 2015   
10841       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
1                   1.0.0        4.0.3 and up  
2                   2.0.0        4.0.3 and up  
3                   1.2.4        4.0.3 and up  
4      Varies with device          4.2 and up  
5                     1.1          4.4 and up  
...                   ...                 ...  
10837                1.48          4.1 and up  
10838                 1.0          4.1 and up  
10839                 1.0          2.2 and up  
10840  Varies with device  Varies with device  
10841  Varies with device  Varies with device  

[9659 rows x 13 columns]

Our firm only build apps that are free to download and install. Therefore, recreate the dataframe only with the apps that are free.

In [19]:
# Drop the rows with non-free apps since we only deal with free apps.
df2 = df2[df2['Price']=='0']

In [20]:
# Reset the index
df2.reset_index(drop=True, inplace=True)
df2

App             Category  \
0        Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                   Coloring book moana       ART_AND_DESIGN   
2     U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                 Sketch - Draw & Paint       ART_AND_DESIGN   
4                 Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                 ...                  ...   
8898                                   Sya9a Maroc - FR               FAMILY   
8899                   Fr. Mike Schmitz Audio Teachings               FAMILY   
8900                             Parkinson Exercices FR              MEDICAL   
8901                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
8902      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

     Rating Reviews                Size     Installs  Type Price  \
0       4.1     159                 19M      10,000+  Free     0   
1       3.9     967                 14M     500,000+  Free     0   
2       4.7   87510                8.7M   5,000,000+  Free     0   
3       4.5  215644                 25M  50,000,000+  Free     0   
4       4.3     967                2.8M     100,000+  Free     0   
...     ...     ...                 ...          ...   ...   ...   
8898    4.5      38                 53M       5,000+  Free     0   
8899    5.0       4                3.6M         100+  Free     0   
8900    NaN       3                9.5M       1,000+  Free     0   
8901    4.5     114  Varies with device       1,000+  Free     0   
8902    4.5  398307                 19M  10,000,000+  Free     0   

     Content Rating                     Genres      Last Updated  \
0          Everyone               Art & Design   January 7, 2018   
1          Everyone  Art & Design;Pretend Play  January 15, 2018   
2          Everyone               Art & Design    August 1, 2018   
3              Teen               Art & Design      June 8, 2018   
4          Everyone    Art & Design;Creativity     June 20, 2018   
...             ...                        ...               ...   
8898       Everyone                  Education     July 25, 2017   
8899       Everyone                  Education      July 6, 2018   
8900       Everyone                    Medical  January 20, 2017   
8901     Mature 17+          Books & Reference  January 19, 2015   
8902       Everyone                  Lifestyle     July 25, 2018   

             Current Ver         Android Ver  
0                  1.0.0        4.0.3 and up  
1                  2.0.0        4.0.3 and up  
2                  1.2.4        4.0.3 and up  
3     Varies with device          4.2 and up  
4                    1.1          4.4 and up  
...                  ...                 ...  
8898                1.48          4.1 and up  
8899                 1.0          4.1 and up  
8900                 1.0          2.2 and up  
8901  Varies with device  Varies with device  
8902  Varies with device  Varies with device  

[8903 rows x 13 columns]

### Removing non-English applications

Since our firm only use English for the apps we develop, and we'd like to analyze only the apps that are directed toward an English-speaking user, we have to remove the non-English based applications. 

We would like to remove each app with a name containing a symbol that is not commonly used in English text — English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;), and other symbols (+, *, /).

Behind the scenes, each character we use in a string has a corresponding number associated with it.

We can get the corresponding number of each character using the ord().

The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127, according to the ASCII (American Standard Code for Information Interchange) system. Based on this number range, we can build a function that detects whether a character belongs to the set of common English characters or not. If the number is equal to or less than 127, then the character belongs to the set of common English characters.

Use indexing to select an individual character from strings, and we can also iterate on the string using a for loop.

However, there are some strings that includes emojis while rest of the letters are English. Therfore to minimize the data loss, only drop the rows that contains more than 3 non-English letter.

Define a function that iterate over given strings and return True if more than 3 letters in the string are non-English.

In [21]:
def iseng(string):
    num = 0
    for i in range(len(string)):
        if ord(string[i]) > 127:
            num += 1
        if num > 3:
            return True


Drop the rows with more than 3 non-English letters in the name of applications from Apple Store Dataset.

In [22]:
for row in df.itertuples():
    if iseng(str(row[2])) == True:
        idx = row[0]       
        df.drop(index=idx, inplace=True, axis=0)
df.reset_index(inplace=True, drop=True)

/Users/kitaeklee/opt/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py:4102: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  errors=errors,


In [23]:
df

id                                         track_name  \
0      281796108                          Evernote - stay organized   
1      281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   
2      282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   
3      282935706                                              Bible   
4      283646709             PayPal - Send and request money safely   
...          ...                                                ...   
3215  1186384912       Demolition Derby Virtual Reality (VR) Racing   
3216  1187617475                                              Kubik   
3217  1187682390                                  VR Roller-Coaster   
3218  1187838770          VR Roller Coaster World - Virtual Reality   
3219  1188375727                       Escape the Sweet Shop Series   

     size_bytes currency price rating_count_tot rating_count_ver user_rating  \
0     158578688      USD     0           161065               26           4   
1     100524032      USD     0           188583             2822         3.5   
2     128512000      USD     0           262241              649           4   
3      92774400      USD     0           985920             5320         4.5   
4     227795968      USD     0           119487              879           4   
...         ...      ...   ...              ...              ...         ...   
3215  168774656      USD     0               18               18           4   
3216  126644224      USD     0              142               75         4.5   
3217  120760320      USD     0               30               30         4.5   
3218   97235968      USD     0               85               32         4.5   
3219   90898432      USD     0                3                3           5   

     user_rating_ver     ver cont_rating   prime_genre sup_devices.num  \
0                3.5   8.2.2          4+  Productivity              37   
1                4.5   5.0.0          4+       Weather              37   
2                4.5  5.10.0         12+      Shopping              37   
3                  5   7.5.1          4+     Reference              37   
4                4.5  6.12.0          4+       Finance              37   
...              ...     ...         ...           ...             ...   
3215               4   1.0.0         12+         Games              38   
3216             4.5     1.3          4+         Games              38   
3217             4.5     0.9          4+         Games              38   
3218             4.5  1.0.15         12+         Games              38   
3219               5     1.0          4+         Games              40   

     ipadSc_urls.num lang.num vpp_lic  
0                  5       23       1  
1                  5        3       1  
2                  5        9       1  
3                  5       45       1  
4                  0       19       1  
...              ...      ...     ...  
3215               4        1       1  
3216               5        1       1  
3217               0        1       1  
3218               0        2       1  
3219               0        2       1  

[3220 rows x 16 columns]

Drop the rows with more than 3 non-English letters in the name of applications from Google Playstore Dataset.

In [24]:
for row in df2.itertuples():
    if iseng(str(row[1])) == True:
        idx = row[0]       
        df2.drop(index=idx, inplace=True, axis=0)
df2.reset_index(inplace=True, drop=True)

In [25]:
df2

App             Category  \
0        Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                   Coloring book moana       ART_AND_DESIGN   
2     U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                 Sketch - Draw & Paint       ART_AND_DESIGN   
4                 Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                 ...                  ...   
8857                                   Sya9a Maroc - FR               FAMILY   
8858                   Fr. Mike Schmitz Audio Teachings               FAMILY   
8859                             Parkinson Exercices FR              MEDICAL   
8860                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
8861      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

     Rating Reviews                Size     Installs  Type Price  \
0       4.1     159                 19M      10,000+  Free     0   
1       3.9     967                 14M     500,000+  Free     0   
2       4.7   87510                8.7M   5,000,000+  Free     0   
3       4.5  215644                 25M  50,000,000+  Free     0   
4       4.3     967                2.8M     100,000+  Free     0   
...     ...     ...                 ...          ...   ...   ...   
8857    4.5      38                 53M       5,000+  Free     0   
8858    5.0       4                3.6M         100+  Free     0   
8859    NaN       3                9.5M       1,000+  Free     0   
8860    4.5     114  Varies with device       1,000+  Free     0   
8861    4.5  398307                 19M  10,000,000+  Free     0   

     Content Rating                     Genres      Last Updated  \
0          Everyone               Art & Design   January 7, 2018   
1          Everyone  Art & Design;Pretend Play  January 15, 2018   
2          Everyone               Art & Design    August 1, 2018   
3              Teen               Art & Design      June 8, 2018   
4          Everyone    Art & Design;Creativity     June 20, 2018   
...             ...                        ...               ...   
8857       Everyone                  Education     July 25, 2017   
8858       Everyone                  Education      July 6, 2018   
8859       Everyone                    Medical  January 20, 2017   
8860     Mature 17+          Books & Reference  January 19, 2015   
8861       Everyone                  Lifestyle     July 25, 2018   

             Current Ver         Android Ver  
0                  1.0.0        4.0.3 and up  
1                  2.0.0        4.0.3 and up  
2                  1.2.4        4.0.3 and up  
3     Varies with device          4.2 and up  
4                    1.1          4.4 and up  
...                  ...                 ...  
8857                1.48          4.1 and up  
8858                 1.0          4.1 and up  
8859                 1.0          2.2 and up  
8860  Varies with device  Varies with device  
8861  Varies with device  Varies with device  

[8862 rows x 13 columns]

Check if Applications like the ones below are dropped from the dataset.
- 'Docs To Go™ Free Office Suite'
- 'Instachat 😜'
- '爱奇艺PPS -《欢乐颂2》电视剧热播'


In [26]:
for row in df2.itertuples():
    if str(row[2]) == 'Docs To Go™ Free Office Suite':
        print("yes")
    elif str(row[2]) == 'Instachat 😜':
        print("yes2")
    elif str(row[2]) == '爱奇艺PPS -《欢乐颂2》电视剧热播':
        print("yes3")


Based on the code above, there are no such applications left in the datset.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets.

Let's begin the analysis by getting a sense of what are the most common genres for each market. For this, we'll need to build frequency tables for a few columns in our data sets.

### Most common apps by genre

First create dictionaries for each categories in each Genre columns from each dataset.

- Dictionary of genres of Apple Store dataset.

In [27]:
dict1={}
for genre in df['prime_genre']:
    if genre in dict1:
        dict1[genre] += 1
    if genre not in dict1:
        dict1[genre] = 1
dict1

{'Productivity': 56,
 'Weather': 28,
 'Shopping': 84,
 'Reference': 18,
 'Finance': 36,
 'Music': 66,
 'Utilities': 81,
 'Travel': 40,
 'Social Networking': 106,
 'Sports': 69,
 'Health & Fitness': 65,
 'Games': 1872,
 'Food & Drink': 26,
 'News': 43,
 'Book': 14,
 'Photo & Video': 160,
 'Entertainment': 254,
 'Business': 17,
 'Lifestyle': 51,
 'Education': 118,
 'Navigation': 6,
 'Medical': 6,
 'Catalogs': 4}

- Two dictionaries of two genre columns of Google Playstore Dataset. 

In [28]:
dict2={}
for genre in df2['Category']:
    if genre in dict2:
        dict2[genre] += 1
    if genre not in dict2:
        dict2[genre] = 1

dict2

{'ART_AND_DESIGN': 60,
 'AUTO_AND_VEHICLES': 82,
 'BEAUTY': 53,
 'BOOKS_AND_REFERENCE': 190,
 'BUSINESS': 407,
 'COMICS': 55,
 'COMMUNICATION': 287,
 'DATING': 165,
 'EDUCATION': 114,
 'ENTERTAINMENT': 100,
 'EVENTS': 63,
 'FINANCE': 328,
 'FOOD_AND_DRINK': 110,
 'HEALTH_AND_FITNESS': 273,
 'HOUSE_AND_HOME': 74,
 'LIBRARIES_AND_DEMO': 83,
 'LIFESTYLE': 346,
 'GAME': 875,
 'FAMILY': 1635,
 'MEDICAL': 312,
 'SOCIAL': 236,
 'SHOPPING': 199,
 'PHOTOGRAPHY': 261,
 'SPORTS': 301,
 'TRAVEL_AND_LOCAL': 207,
 'TOOLS': 748,
 'PERSONALIZATION': 294,
 'PRODUCTIVITY': 345,
 'PARENTING': 58,
 'WEATHER': 71,
 'VIDEO_PLAYERS': 158,
 'NEWS_AND_MAGAZINES': 248,
 'MAPS_AND_NAVIGATION': 124}

In [29]:
dict3={}

for genre in df2['Genres']:
    if genre in dict3:
        dict3[genre] += 1
    if genre not in dict3:
        dict3[genre] = 1

dict3

{'Art & Design': 53,
 'Art & Design;Pretend Play': 1,
 'Art & Design;Creativity': 6,
 'Art & Design;Action & Adventure': 1,
 'Auto & Vehicles': 82,
 'Beauty': 53,
 'Books & Reference': 190,
 'Business': 407,
 'Comics': 54,
 'Comics;Creativity': 1,
 'Communication': 287,
 'Dating': 165,
 'Education;Education': 30,
 'Education': 474,
 'Education;Creativity': 4,
 'Education;Music & Video': 3,
 'Education;Action & Adventure': 3,
 'Education;Pretend Play': 5,
 'Education;Brain Games': 3,
 'Entertainment': 538,
 'Entertainment;Music & Video': 15,
 'Entertainment;Brain Games': 7,
 'Entertainment;Creativity': 3,
 'Events': 63,
 'Finance': 328,
 'Food & Drink': 110,
 'Health & Fitness': 273,
 'House & Home': 74,
 'Libraries & Demo': 83,
 'Lifestyle': 345,
 'Lifestyle;Pretend Play': 1,
 'Adventure;Action & Adventure': 3,
 'Arcade': 164,
 'Casual': 156,
 'Card': 40,
 'Casual;Pretend Play': 21,
 'Action': 275,
 'Strategy': 81,
 'Puzzle': 100,
 'Sports': 307,
 'Music': 18,
 'Word': 23,
 'Racing': 8

- Function that return frequency values and function that display frequency in percentage(%) in descending order.

In [30]:
def freq_table(dataset):
    dictionary={}
    total = 0
    for value in dataset:
        dictionary[value] = dataset[value]
        total += dataset[value]
    for value in dictionary:
        dictionary[value] = dictionary[value]/total
    return dictionary


In [31]:
def display_table(dataset):
    dataset = freq_table(dataset)
    # Create a list of tuples sorted by index 1 i.e. value field     
    listofTuples = sorted(dataset.items() , reverse=True, key=lambda x: x[1])
    # Iterate over the sorted sequence
    for elem in listofTuples :
        print(elem[0] , " : " , '{:.2%}'.format(elem[1]))   

- Frequency of apps sold in Apple Store

In [32]:
display_table(dict1)

Games  :  58.14%
Entertainment  :  7.89%
Photo & Video  :  4.97%
Education  :  3.66%
Social Networking  :  3.29%
Shopping  :  2.61%
Utilities  :  2.52%
Sports  :  2.14%
Music  :  2.05%
Health & Fitness  :  2.02%
Productivity  :  1.74%
Lifestyle  :  1.58%
News  :  1.34%
Travel  :  1.24%
Finance  :  1.12%
Weather  :  0.87%
Food & Drink  :  0.81%
Reference  :  0.56%
Business  :  0.53%
Book  :  0.43%
Navigation  :  0.19%
Medical  :  0.19%
Catalogs  :  0.12%


In the Apple Store, **Games genre** is dominating over all the other genres by 58%, followed by Entertainment by 7.89%, Photo & Video by 4.97%, Education by 3.66% and Social Networking by 3.29%.

Although Game genre applications are the majority, it does not imply that Game genre applications are the genre with the most installed which reflects how many users are actually there.

According to Top 5 genres, the most common applications are for entertainment (Games, Entertainment, Photo & Video, Social Networking, Shopping, etc.). Presumably Top 5 genres except Education seems applications for relatively young people. Therefore, I can assume most of users are relatively young. I believe the reason why the fourth most common application is Education is because younger people tend to be more in need of Education or many of them could be presumably are school-related applications.

- Frequency of apps sold in Google Playstore

In [33]:
display_table(dict2)

FAMILY  :  18.45%
GAME  :  9.87%
TOOLS  :  8.44%
BUSINESS  :  4.59%
LIFESTYLE  :  3.90%
PRODUCTIVITY  :  3.89%
FINANCE  :  3.70%
MEDICAL  :  3.52%
SPORTS  :  3.40%
PERSONALIZATION  :  3.32%
COMMUNICATION  :  3.24%
HEALTH_AND_FITNESS  :  3.08%
PHOTOGRAPHY  :  2.95%
NEWS_AND_MAGAZINES  :  2.80%
SOCIAL  :  2.66%
TRAVEL_AND_LOCAL  :  2.34%
SHOPPING  :  2.25%
BOOKS_AND_REFERENCE  :  2.14%
DATING  :  1.86%
VIDEO_PLAYERS  :  1.78%
MAPS_AND_NAVIGATION  :  1.40%
EDUCATION  :  1.29%
FOOD_AND_DRINK  :  1.24%
ENTERTAINMENT  :  1.13%
LIBRARIES_AND_DEMO  :  0.94%
AUTO_AND_VEHICLES  :  0.93%
HOUSE_AND_HOME  :  0.84%
WEATHER  :  0.80%
EVENTS  :  0.71%
ART_AND_DESIGN  :  0.68%
PARENTING  :  0.65%
COMICS  :  0.62%
BEAUTY  :  0.60%


The most common genres categorized by Category are Family by 18.45% followed by Game by 9.87%, Tools by 8.44%, Business by 4.59% and Lifestyle by 3.90%. The first two categories seems like they are for entertainment while the next three are for practical purposes (Tools, Business, Lifestyle).

In [34]:
display_table(dict3)

Tools  :  8.43%
Entertainment  :  6.07%
Education  :  5.35%
Business  :  4.59%
Lifestyle  :  3.89%
Productivity  :  3.89%
Finance  :  3.70%
Medical  :  3.52%
Sports  :  3.46%
Personalization  :  3.32%
Communication  :  3.24%
Action  :  3.10%
Health & Fitness  :  3.08%
Photography  :  2.95%
News & Magazines  :  2.80%
Social  :  2.66%
Travel & Local  :  2.32%
Shopping  :  2.25%
Books & Reference  :  2.14%
Simulation  :  2.04%
Dating  :  1.86%
Arcade  :  1.85%
Video Players & Editors  :  1.77%
Casual  :  1.76%
Maps & Navigation  :  1.40%
Food & Drink  :  1.24%
Puzzle  :  1.13%
Racing  :  0.99%
Libraries & Demo  :  0.94%
Role Playing  :  0.94%
Auto & Vehicles  :  0.93%
Strategy  :  0.91%
House & Home  :  0.84%
Weather  :  0.80%
Events  :  0.71%
Adventure  :  0.68%
Comics  :  0.61%
Art & Design  :  0.60%
Beauty  :  0.60%
Parenting  :  0.50%
Card  :  0.45%
Casino  :  0.43%
Trivia  :  0.42%
Educational;Education  :  0.39%
Board  :  0.37%
Educational  :  0.37%
Education;Education  :  0.34%
Wor

The most common genres categorized by Genres are Tools by 8.43% followed by Entertainment by 6.07%, Education by 5.35%, Business by 4.59% and Lifestyle by 3.89%. 

While the first two most common categories were for entertainment, the most common applications when we categorized by genres are designed for practical purposes (Tools, Business, Education, Productivity, Lifestyle).

Google Playstore data categorized applications in two different categories which are "Category" and "Genres". 

In [35]:
df2[df2['Genres']=='Entertainment']['Category'].unique()

array(['ENTERTAINMENT', 'FAMILY'], dtype=object)

In [36]:
df2[df2['Category']=='FAMILY']['Genres'].unique()

array(['Casual;Brain Games', 'Educational;Creativity',
       'Puzzle;Brain Games', 'Educational;Education',
       'Education;Creativity', 'Educational;Brain Games',
       'Educational;Pretend Play', 'Education;Education',
       'Casual;Action & Adventure', 'Entertainment;Education',
       'Entertainment;Brain Games', 'Casual;Education',
       'Casual;Pretend Play', 'Casual;Creativity', 'Music;Music & Video',
       'Simulation;Action & Adventure', 'Racing;Action & Adventure',
       'Entertainment;Music & Video', 'Arcade;Pretend Play',
       'Action;Action & Adventure', 'Education;Pretend Play',
       'Adventure;Action & Adventure', 'Role Playing;Action & Adventure',
       'Simulation;Pretend Play', 'Puzzle;Creativity',
       'Sports;Action & Adventure', 'Educational;Action & Adventure',
       'Arcade;Action & Adventure', 'Entertainment;Action & Adventure',
       'Puzzle;Action & Adventure', 'Education;Action & Adventure',
       'Strategy;Action & Adventure', 'Music & Audi

In [37]:
df2['Genres'].unique()

array(['Art & Design', 'Art & Design;Pretend Play',
       'Art & Design;Creativity', 'Art & Design;Action & Adventure',
       'Auto & Vehicles', 'Beauty', 'Books & Reference', 'Business',
       'Comics', 'Comics;Creativity', 'Communication', 'Dating',
       'Education;Education', 'Education', 'Education;Creativity',
       'Education;Music & Video', 'Education;Action & Adventure',
       'Education;Pretend Play', 'Education;Brain Games', 'Entertainment',
       'Entertainment;Music & Video', 'Entertainment;Brain Games',
       'Entertainment;Creativity', 'Events', 'Finance', 'Food & Drink',
       'Health & Fitness', 'House & Home', 'Libraries & Demo',
       'Lifestyle', 'Lifestyle;Pretend Play',
       'Adventure;Action & Adventure', 'Arcade', 'Casual', 'Card',
       'Casual;Pretend Play', 'Action', 'Strategy', 'Puzzle', 'Sports',
       'Music', 'Word', 'Racing', 'Casual;Creativity',
       'Casual;Action & Adventure', 'Simulation', 'Adventure', 'Board',
       'Trivia', 'Role 

In [38]:
df2['Category'].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION'],
      dtype=object)

There are differences between two since, as we can see from the code above as an example, even though some applications are both categorized into same Genres, that doesn't mean they are same type of category. For example, 'Entertainment' Genres are categorized either into 'Entertainment' or 'Family'. 

"Genres" are more in detail while 'Category' is showing the applications in broader range.

Frequency table of applications can show what are the most common genres that developers are making but we can't be sure that it reflects those are what most users use and the most downloaded.

### Most popular apps by genre

To find out what genres are the most popular (have the most users), we should calculate the average number of installs for each app genre. 

For the Google Play data set, we can find this information in the Installs column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

##### Apple Store

In [105]:
df_pop_apple = df.loc[:, ['prime_genre', 'rating_count_tot']].groupby('prime_genre').sum()
df_pop_apple.reset_index(inplace=True)

#df_pop_apple['rating_count_tot'] = pd.to_numeric(df_pop_apple['rating_count_tot'], errors='coerce')
#df_pop_apple = df_pop_apple.sort_values(by='rating_count_tot', ascending=False)

df_pop_apple['num_of_apps'] = 



prime_genre                                   rating_count_tot
0                Book       25207665450105274879045119739247829840620900
1            Business  3868111502008258623832424744200694053464392117...
2            Catalogs                                    431133452132027
3           Education  1231901348257821356771975206498117774188865160...
4       Entertainment  8100618229178746886131178892425817105518884987...
5             Finance  1194871197732332702329404882243064199467717516...
6        Food & Drink  1139361093492230385629258624297502056659519113...
7               Games  4231618008750880818731553465114096556582445118...
8    Health & Fitness  5795373835339691045395362013683374691029850770...
9           Lifestyle  3429691926125521011111948939108310047760490498...
10            Medical                                   0134111982911040
11              Music  1126879402925293228467748905386487856313574426...
12         Navigation                          3450461281118753582154911
13               News  4840729107112118328811151128863540581169127316...
14      Photo & Video  3972251427283815654216155828189647810286836683...
15       Productivity  1610651232152204957829110168396381021359623249...
16          Reference  9859202000475417518418168492678612122762001417...
17           Shopping  2622411081311263128042417906263538888541863450...
18  Social Networking  2974676718563735193342936065928758916496323201...
19             Sports  1379515750059639148160290996541922095310472375...
20             Travel  5619422388544618524839751226830552312614946680...
21          Utilities  4794406501610850719126838130450165226697111099...
22            Weather  1885834956261442141501582086482453608123331284...

In [132]:
df_pop_apple['rating_count_tot'].astype('int')

OverflowError: Python int too large to convert to C long

In [127]:
# Add new column "number of applications"
for value in df_pop_apple['prime_genre']:
    df_pop_apple['num_of_apps'][df['prime_genre']==value] = df['prime_genre'][df['prime_genre'] == value].count()

/Users/kitaeklee/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [130]:
df_pop_apple['avg_num_rating'] = (df_pop_apple['rating_count_tot'] / df_pop_apple['num_of_apps'])

TypeError: unsupported operand type(s) for /: 'str' and 'int'

In [131]:
df_pop_apple

prime_genre                                   rating_count_tot  \
0                Book       25207665450105274879045119739247829840620900   
1            Business  3868111502008258623832424744200694053464392117...   
2            Catalogs                                    431133452132027   
3           Education  1231901348257821356771975206498117774188865160...   
4       Entertainment  8100618229178746886131178892425817105518884987...   
5             Finance  1194871197732332702329404882243064199467717516...   
6        Food & Drink  1139361093492230385629258624297502056659519113...   
7               Games  4231618008750880818731553465114096556582445118...   
8    Health & Fitness  5795373835339691045395362013683374691029850770...   
9           Lifestyle  3429691926125521011111948939108310047760490498...   
10            Medical                                   0134111982911040   
11              Music  1126879402925293228467748905386487856313574426...   
12         Navigation                          3450461281118753582154911   
13               News  4840729107112118328811151128863540581169127316...   
14      Photo & Video  3972251427283815654216155828189647810286836683...   
15       Productivity  1610651232152204957829110168396381021359623249...   
16          Reference  9859202000475417518418168492678612122762001417...   
17           Shopping  2622411081311263128042417906263538888541863450...   
18  Social Networking  2974676718563735193342936065928758916496323201...   
19             Sports  1379515750059639148160290996541922095310472375...   
20             Travel  5619422388544618524839751226830552312614946680...   
21          Utilities  4794406501610850719126838130450165226697111099...   
22            Weather  1885834956261442141501582086482453608123331284...   

    num_of_apps  
0            56  
1            28  
2            84  
3            18  
4            36  
5            66  
6            81  
7            36  
8            40  
9           106  
10           40  
11           66  
12           69  
13           65  
14         1872  
15           56  
16          106  
17         1872  
18           66  
19           66  
20           66  
21           40  
22         1872